src/main/scala/org/broadinstitute/dsde/firecloud/core/AgoraPermissionHandler.scala

```diff
@@ -0,0 +1,307 @@
+    ownerAllList=(ownerAllList:+"All")
+    val agoraAllPerm=AgoraPermission(
+      user=Some("all_user"),
+      roles=Some(ownerAllList))
+    val FCAll=agoraAllPerm.toFireCloudPermission
+    println("For Agora ALL permission "+agoraAllPerm+" the FC is "+FCAll)
+
+  return true
+  }
+
+
+  def testTranslations() = {
+    println("\n\n\n*****************************")
+    println("NOW TESTING AGORA->FC TRANSLATION")
+    this.testFromAgoraToFC()
+    println("\n\n\n*****************************")
+    println("NOW TESTING RANDOM AGORA->FC TRANSLATION")
+    this.testFromRandomAgoraToFC()
+    println("\n\n\n*****************************")
+    println("NOW TESTING FC->AGORA TRANSLATION")
+    this.testFromFCToAgora()
+
+  }
+
+  // convenience method to translate a FireCloudPermission object to an AgoraPermission object
+  def toAgoraPermission(fireCloudPermission: FireCloudPermission):AgoraPermission = {
+    fireCloudPermission.user match {
+      case None => {
+        // TODO: should throw an error if user is not specified
+        AgoraPermission(None, Some(List.empty))
+      }
+      case Some(u) => AgoraPermission(Some(u), Some(toAgoraRoles(fireCloudPermission.role)))
+    }
+  }
+
+  // convenience method to translate an AgoraPermission to a FireCloudPermission object
+  def toFireCloudPermission(agoraPermission: AgoraPermission):FireCloudPermission = {
+    agoraPermission.user match {
+      case None => {
+        // TODO: should throw an error if user is not specified
+        FireCloudPermission(None, Some("NO ACCESS"))
+      }
+      case Some(u) => FireCloudPermission(Some(u), Some(toFireCloudRole(agoraPermission.roles)))
+    }
+  }
+
+
+  def hasValidUserAndRole(fireCloudPermission: FireCloudPermission):Boolean = {
+    val vRole=this.hasValidRole(fireCloudPermission)
+    val vUser=this.hasValidUser(fireCloudPermission)
+    return (vRole && vUser)
+  }
+
+
+  def hasValidUser(fireCloudPermission: FireCloudPermission):Boolean = {
+    fireCloudPermission.user match {
+      case Some("") => false
+      case None => false
+      //any non-empty user is okay!
+      case _ => true
+    }
+  }
+
+
+  def hasValidRole(fireCloudPermission: FireCloudPermission):Boolean = {
+    fireCloudPermission.role match {
+      case Some("NO ACCESS") => true
+      case Some("OWNER") => true
+      case Some("READER") => true
+      //anything that's one of the above-3 is good, anything else is bad
+      case _ => false
+    }
+  }
+
+
+  // translation between a FireCloud role and a list of Agora roles
+  def toAgoraRoles(fireCloudRole:Option[String]) = {
+    fireCloudRole match {
+      case None => List("Nothing")
+      case Some("OWNER") => List("Read","Write","Create","Redact","Manage") // Could use "All" instead but this is more precise
+      case Some("READER") => List("Read")
+      case Some("NO ACCESS") => List("Nothing")
+      case _ => List("Nothing") // TODO: throw an exception instead? Log something?
```

Comment:
I don't understand "No need to introduce an exception - Nothing should be good."